packages/flutter_tools/lib/src/base/process_manager.dart

```diff
@@ -7,14 +7,21 @@ import 'dart:convert';
 import 'dart:io';
 
 import 'package:archive/archive.dart';
+import 'package:collection/collection.dart';
 import 'package:intl/intl.dart';
 import 'package:path/path.dart' as path;
 
 import 'context.dart';
+import 'os.dart';
 import 'process.dart';
 
 ProcessManager get processManager => context[ProcessManager];
 
+typedef bool ListsEqual<T>(List<T> list1, List<T> list2);
+
+const String _kManifestName = 'MANIFEST.txt';
+final ListsEqual<String> kListsEqual = const ListEquality<String>().equals;
```

Comment:
rather than using collection.dart, just have a function somewhere in flutter_tool that does this (if you use it a lot) or just inline it (if not).

```dart
if (list1.length != list2.length)
  return false;
for (int index = 0; index < list1.length; index += 1) {
  if (list1[index] != list2[index)
    return false;
}
return true;
```